src/averageAges.js

```diff
@@ -10,16 +10,24 @@
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
- * @param {number} century - optional
+ * @param {number} century
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let avarageAge = 0;
+
+  century
+    ? avarageAge = people.filter(person => person.sex === 'm'
+      && Math.ceil(person.died / 100) === century)
+      .reduce((a, b) => a + b.died - b.born, 0) / people
+      .filter(person => person.sex === 'm'
+      && Math.ceil(person.died / 100) === century).length
+    : avarageAge = people.filter(person => person.sex === 'm')
+      .reduce((a, b) => a + b.died - b.born, 0) / people
+      .filter(person => person.sex === 'm').length;
+
+  return avarageAge;
 }
```

Comment:
Looks too complicated, let's improve your code a bit:

Here are an example:

```
  const men= people.filter(person =>
    century
      ? person.sex === 'm' && Math.ceil(person.died / 100) === century
      : person.sex === 'm'
  );
```
  
  And then result of reduce divided by men.length
  
  Same for women and averageAge.
  
  You can also move reduce logic to separate function and reuse it